src/create-with-theme.js

```diff
@@ -1,43 +1,35 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import channel from './channel';
+import createThemeListener from './create-theme-listener';
 
 const getDisplayName = Component =>
   Component.displayName || Component.name || 'Component';
 
 export default function createWithTheme(CHANNEL = channel) {
+  const themeListener = createThemeListener(CHANNEL);
   return Component =>
     class WithTheme extends React.Component {
       static displayName = `WithTheme(${getDisplayName(Component)})`;
-      state = { theme: {} };
-      setTheme = theme => this.setState({ theme });
+      static contextTypes = themeListener.contextTypes;
 
-      static contextTypes = {
-        [CHANNEL]: PropTypes.object.isRequired,
-      };
+      constructor(props) {
+        super(props);
+        this.state = { theme: {} };
```

Comment:
i dunno, the same can easily achieved by `defaultProps` of Component being enhanced